server/routes/v1/admin.ts

```diff
@@ -0,0 +1,124 @@
+import { Request, Response } from 'express'
+import QueryString from 'qs'
+import { LogType } from '../../../types/interfaces'
+import { getLogs, getLogType } from '../../services/logs'
+import { BadReq } from '../../utils/result'
+import { ensureUserRole } from '../../utils/user'
+
+function parseDateQuery(
+  queryDate: string | string[] | QueryString.ParsedQs | QueryString.ParsedQs[] | undefined,
+  fallback: Date
+): Date {
+  if (typeof queryDate !== 'string') {
+    return fallback
+  }
+
+  const date = Date.parse(queryDate)
+
+  if (Number.isNaN(date)) {
+    return fallback
+  }
+
+  return new Date(date)
+}
+
+function queryStringArrayTypeGuard(query: unknown): query is QueryString.ParsedQs[] {
+  return Array.isArray(query) && typeof query[0] === 'object'
+}
+
+function parseQueryArray(
+  property: string,
+  query: string | string[] | QueryString.ParsedQs | QueryString.ParsedQs[] | undefined
+): string[] {
+  if (!query) {
+    return []
+  }
+
+  if ((!Array.isArray(query) && typeof query !== 'string') || queryStringArrayTypeGuard(query)) {
+    throw BadReq({ [property]: query }, `Should not pass an object to ${property}`)
+  }
+
+  if (!Array.isArray(query)) {
+    return [query]
+  }
+
+  return query
+}
+
+function parseString(
+  property: string,
+  query: string | string[] | QueryString.ParsedQs | QueryString.ParsedQs[] | undefined
+): string | undefined {
+  if (query !== undefined && typeof query !== 'string') {
+    throw BadReq({ [property]: query }, `Must pass a string to ${property}`)
+  }
+
+  return query
+}
+
+export const getApplicationLogs = [
+  ensureUserRole(['user', 'admin']),
+  async (req: Request, res: Response) => {
+    const after = parseDateQuery(req.query.after, new Date(0))
+    const before = parseDateQuery(req.query.before, new Date(Date.now()))
+    const level = parseInt(parseString('level', req.query.level) || '0', 10)
+
+    if (Number.isNaN(level)) {
+      throw BadReq({ level: req.query.level }, 'Level cannot be passed as number')
```

Comment:
Would parsed make more sense here?

Reply:
I think that's what the intention was. I'll update under that assumption.